### PR TITLE
Improve scale pad for more aspect mixes and fix tests

### PR DESCRIPTION
**Pull request by maxwellimpact**
This changes how we pad/scale videos by scaling the video before padding. Otherwise pad will error out. Especially if the aspect ratio is wider and shorter than the output.

This also fixes all of the tests and improves on how they are written to make it easier to update for the next time.

### CODE REVIEW

File: test/test-ffmpegCommandService.js [hunk 1]
```diff
@@ -525,10 +693,63 @@ describe('FfmpegCommandService', function() {
...
+                    `-i ${_path('./fixtures/assets1/sample2.mp4')} ` +
+                    '-f lavfi -i color=c=555555:size=1920x1080 ' +
+                    '-f lavfi -i color=c=ff0000:size=1920x1080 ' +
+                    '-loop 1 ' +
+                    `-i ${_path('./fixtures/assets1/logo.png')} ` +
+                    '-loop 1 ' +
+                    `-i ${_path('./fixtures/assets1/logo.png')} ` +
+                    `-i ${_path('./fixtures/assets1/music.mp3')} ` +
+                    `-i ${_path('./fixtures/assets1/music2.mp3')} ` +
+                    '-filter_complex "' +
```
Comment (maxwellimpact) [inline test/test-ffmpegCommandService.js:711]: As the commands got bigger I tried to use indentation to help clarify what was going on. Hopefully this makes it easier for the next person to edit.

File: lib/ffmpegCommandService.js [hunk 2]
```diff
@@ -374,13 +374,13 @@ class FfmpegCommandService {
...
             filter += ` [${i}:v]`; // start
 
+            // Scale videos to the same size (setsar=1 recommended to coax aspect ratios, may not be needed anymore)
+            filter += ` scale=${outWidth}:${outHeight}:force_original_aspect_ratio=decrease, setsar=1,`;
+
             // Force padding for videos not in the correct aspect ratio ih=inputHeight, oh=outputHeight, etc
             // See examples https://ffmpeg.org/ffmpeg-filters.html#pad-1
             filter += ` pad=ih*${ratio}/sar:ih:(ow-iw)/2:(oh-ih)/2,`;
 
-            // Scale videos to the same size (setsar=1 recommended to coax aspect ratios, may not be needed anymore)
```
Comment (alexgandy) [inline lib/ffmpegCommandService.js:381]: Might want to remove the "might not be needed anymore" comment

File: test/test-ffmpegCommandService.js [hunk 3]
```diff
@@ -5,6 +5,10 @@ const Path = require('path');
 
 const FfmpegCommandServiceFactory = require('../lib/ffmpegCommandServiceFactory');
 
+function _path(uri) {
```
Comment (alexgandy) [inline test/test-ffmpegCommandService.js:8]: `uri`?

`filePath` maybe?

File: lib/ffmpegCommandService.js (hunk 2 above)
Comment (maxwellimpact) [reply lib/ffmpegCommandService.js:381]: So I'm actually not sure if its needed... I need to revisit. I'll make it a todo.

File: test/test-ffmpegCommandService.js [hunk 4]
```diff
@@ -45,10 +49,23 @@ describe('FfmpegCommandService', function() {
...
                     },
-                    workingDirectory: Path.resolve(__dirname, './fixtures/assets1')
-                });
-
-                Should(command).eql('/usr/local/bin/ffmpeg -i ' + Path.resolve(__dirname, './fixtures/assets1/sample.mp4') + ' -filter_complex "[0:v] trim=start=0:duration=600, setpts=PTS-STARTPTS [v0]; [v0] concat=n=1:v=1:a=0 [v_concat]; [v_concat] scale=1920:1080 [v_scaled]" -map "[v_scaled]" -y ' + Path.resolve(__dirname, './fixtures/assets1/output.mp4'));
+                    workingDirectory: _path('./fixtures/assets1')
+                });
+
+                const expected = '/usr/local/bin/ffmpeg ' +
+                    '-i ' + _path('./fixtures/assets1/sample.mp4') +
```
Comment (alexgandy) [inline test/test-ffmpegCommandService.js:56]: I'd love to store these somewhere else...or create custom matchers for them. This is real dirty.

File: test/test-ffmpegCommandService.js (hunk 1 above)
Comment (alexgandy) [reply test/test-ffmpegCommandService.js:711]: Thanks!

**Review by alexgandy — APPROVED** [review]

**Pull request merged**